src/wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php

```diff
@@ -2054,7 +2054,18 @@ public function prepare_item_for_response( $item, $request ) {
 		}
 
 		if ( rest_is_field_included( 'meta', $fields ) ) {
-			$data['meta'] = $this->meta->get_value( $post->ID, $request );
+			$meta_value = $this->meta->get_value( $post->ID, $request );
+
+			// Ensure empty meta value is returned as an object {} if the schema expects an object
+			// and the context is 'view'. For 'edit' context, it should remain an empty array
+			// to ensure PUT requests expecting an array function correctly.
+			if ( 'view' === $request['context'] && empty( $meta_value ) && is_array( $meta_value ) ) {
```

Comment:
I don't understand why the request context is relevant. If the type is `object`, the value should _always_ be an object, right? What would be the specific reason to not respect the type in `edit` context?